service/history/task/priority_assigner.go

```diff
@@ -69,26 +69,32 @@ func NewPriorityAssigner(
 func (a *priorityAssignerImpl) Assign(
 	queueTask Task,
 ) error {
-	if queueTask.GetQueueType() == QueueTypeReplication {
+	queueType := queueTask.GetQueueType()
+
+	if queueType == QueueTypeReplication {
 		queueTask.SetPriority(task.GetTaskPriority(task.LowPriorityClass, task.DefaultPrioritySubclass))
 		return nil
 	}
 
-	// timer of transfer task, first check if domain is active or not
+	// timer or transfer task, first check if task is active or not and if domain is active or not
+	isActiveTask := queueType == QueueTypeActiveTimer || queueType == QueueTypeActiveTransfer
 	domainName, active, err := a.getDomainInfo(queueTask.GetDomainID())
 	if err != nil {
 		return err
 	}
 
-	if !active {
+	if !isActiveTask && !active {
+		// only assign low priority to tasks submitted by standby processor for standby domains
```

Comment:
nit: list out the four cases and maybe mention which cases are no-op?

Reply:
Done.